packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/marker.dart

```diff
@@ -140,26 +140,38 @@ class Marker implements MapsObject<Marker> {
   /// * reports [onDragEnd] events
   const Marker({
     required this.markerId,
+    this.icon,
     this.alpha = 1.0,
     this.anchor = const Offset(0.5, 1.0),
     this.consumeTapEvents = false,
     this.draggable = false,
     this.flat = false,
-    this.icon = BitmapDescriptor.defaultMarker,
     this.infoWindow = InfoWindow.noText,
     this.position = const LatLng(0.0, 0.0),
     this.rotation = 0.0,
     this.visible = true,
     this.zIndex = 0.0,
     this.onTap,
+    this.count,
+    this.price,
     this.onDrag,
     this.onDragStart,
     this.onDragEnd,
-  }) : assert(alpha == null || (0.0 <= alpha && alpha <= 1.0));
+  }) : assert(0.0 <= alpha && alpha <= 1.0 && (price != null || count != null));
```

Comment:
Won't this break for iOS? Maybe add `|| icon != null`? We could also have an assertion checking that price & count shouldn't be set together

Reply:
A different approach here (not sure if it's better tho) is having a `String label` and a `MarkerLabelType labelTyle` instead (not really sure if we need this enum). `MarkerLabelType` being:

```dart
enum MarkerLabelType { price, count }
```

The good part about it imo is that we won't need those assertions or disclaimers since we'll have a single value to set for the marker's label.

Reply:
A `MarkerType` enum would be useful to distinguish the type of marker that would be used (count or price), or if there would be any new marker in the future (in which we could just add to the enum and pass the value in the `label` argument). Nice idea!